### PR TITLE
docs: add usage of standalone APIs to packages

### LINKED ISSUE #3612: Docs: Document usage of Standalone APIs for each package
### Information

Each package will have a standalone API. We should document how to use the standalone APIs from initial setup to additional usage.

### Documentation page

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

Adds documentation of standalone APIs for each page that needs them.

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #3612

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/effects/index.md [hunk 1]
```diff
@@ -194,7 +194,29 @@ The `EffectsModule.forRoot()` method must be added to your `AppModule` imports e
...
 </div>
 
-Effects start running immediately after the AppModule is loaded to ensure they are listening for all relevant actions as soon as possible.
+### Using the Standalone API
+
+Registering effects can also be done using the standalone APIs if you are bootstrapping an Angular application using standalone features.
+
+<code-example header="main.ts">
+import { bootstrapApplication } from '@angular/platform-browser';
+import { provideStore, provideState } from '@ngrx/store';
```
Comment (cedricduffournet) [inline projects/ngrx.io/content/guide/effects/index.md:203]:
```suggestion
import { provideStore } from '@ngrx/store';
import { provideEffects } from '@ngrx/effects';
```

File: projects/ngrx.io/content/guide/data/index.md [hunk 2]
```diff
@@ -102,6 +102,29 @@ import { entityConfig } from './entity-metadata';
...
+import { provideEntityData, withEffects } from '@ngrx/data';
+
+import { AppComponent } from './app.component';
+import { entityConfig } from './entity-metadata';
+
+bootstrapApplication(AppComponent, {
+  providers: [
+    provideHttpClient(),
+    provideStore(),
+    provideEffects(),
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/data/index.md:122]: This is missing an import (here and in all other snippets)

File: projects/ngrx.io/content/guide/effects/index.md [hunk 3]
```diff
@@ -194,7 +194,29 @@ The `EffectsModule.forRoot()` method must be added to your `AppModule` imports e
...
+
+import { AppComponent } from './app.component';
+
+bootstrapApplication(AppComponent, {
+  providers: [
+    provideStore(),
+    provideEffects(MovieEffects)
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/effects/index.md:210]: `MovieEffects` needs to be imported too

File: projects/ngrx.io/content/guide/effects/index.md [hunk 4]
```diff
@@ -194,7 +194,29 @@ The `EffectsModule.forRoot()` method must be added to your `AppModule` imports e
...
+  ],
+});
+</code-example>
+
+<div class="alert is-important">
+
+Effects start running **immediately** after instantiation to ensure they are listening for all relevant actions as soon as possible. Services used in root-level effects are **not** recommended to be used with services that are used with the `APP_INITIALIZER` token.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/effects/index.md:217]: 👍 good note

File: projects/ngrx.io/content/guide/store-devtools/index.md [hunk 5]
```diff
@@ -34,4 +34,29 @@ import { StoreDevtoolsModule } from '@ngrx/store-devtools';
...
+import { provideStoreDevtools } from '@ngrx/store-devtools';
+
+import { AppComponent } from './app.component';
+
+bootstrapApplication(AppComponent, {
+  providers: [
+    provideStore(),
+    provideStoreDevtools({
+      maxAge: 25, // Retains last 25 states
+      logOnly: !isDevMode(), // Restrict extension to log-only mode
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store-devtools/index.md:53]: `isDevMode` needs to be imported here

File: projects/ngrx.io/content/guide/store/reducers.md [hunk 6]
```diff
@@ -121,7 +121,32 @@ export class AppModule {}
...
+  providers: [
+    provideStore(),
+    provideState({ game: scoreboardReducer })
+  ],
+});
+</code-example>
+
+<div class="alert is-important">
+
+**Note:** Although you can register reducers in the `provideStore()` function, we recommened keeping `provideStore()` empty and using the `provideState()` function to register feature states in the root `providers` array.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/reducers.md:145]: What do you think of creating a ESLint rule for this?

File: projects/ngrx.io/content/guide/store/reducers.md [hunk 7]
```diff
@@ -165,6 +205,26 @@ import { scoreboardFeatureKey, scoreboardReducer } from './reducers/scoreboard.r
 export class ScoreboardModule {}
 </code-example>
 
+### Using the Standalone API
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/reducers.md:208]: Should we also document using standalone with `createFeature`?
https://ngrx.io/guide/store/feature-creators

File: projects/ngrx.io/content/guide/store/reducers.md (hunk 6 above)
Comment (brandonroberts) [reply projects/ngrx.io/content/guide/store/reducers.md:145]: 🤔  I don't think it should be an error, a warning if anything.

File: projects/ngrx.io/content/guide/store/reducers.md (hunk 7 above)
Comment (brandonroberts) [reply projects/ngrx.io/content/guide/store/reducers.md:208]: Good catch

**Review by cedricduffournet — APPROVED** [review]

**Review by timdeschryver — APPROVED** [review]

**Review by markostanimirovic — APPROVED** [review]
🚀

**Pull request merged**